File: Exercism/java/darts/src/main/java/Darts.java
import java.lang.Math;
class Darts {
    private final int score;

    Darts(double x, double y) {
        double l = Math.sqrt(x*x + y*y);
        if        (l <= 1) {
            score = 10;
        } else if (l <= 5) {
            score = 5;
        } else if (l <= 10) {
            score = 1;
        } else {
            score = 0;
        }
    }

    int score() {
        return score;
    }

}

File: Exercism/c/darts/src/darts.c
#include "darts.h"

uint8_t score(coordinate_t landing_position) {
    float x = landing_position.x;
    float y = landing_position.y;
    float r = sqrtf(x*x + y*y);
    
    if      (r <= 1.0) return 10;
    else if (r <= 5.0) return 5;
    else if (r <= 10.0) return 1;
    
    return 0;
}

